README.md

```diff
@@ -443,6 +443,34 @@ private:
 
 Function names are lowercase with underscores between words.
 
+## Header files
+
+### Protecting agains multiple inclusion
+
+Public header files should be protected with a `#ifndef / #define` guard that mimicks the path (excluding
+the initial `include` folder) the and filename, and surrounded with underscores.
+For instance, header file `include/fastdds/dds/core/LoanableArray.hpp` should use `_FASTDDS_DDS_CORE_LOANABLEARRAY_HPP_`.
+To avoid guards duplicity, underscores and any character that cannot be used on a `#define` are forbidden on folder names.
+
+If a filename has any character not usable on a `#define` it is removed when doing the conversion,
+but the presence of such characters is discouraged.
+For instance, header file `include/fastdds/header-one.hpp` would use define `_FASTDDS_HEADERONE_HPP`.
+It is thus forbidden to have two headers on the same folder with names that differ only on special characters (having
+`header-one.hpp` you cannot have `header_one.hpp` or `headerone.hpp`)
+
+### Header contents
+
+A header should only contain one of the following contents:
+
+* the declarartion of a single struct/class, along with related static operators
+* a bunch of constants on a single namespace
+* a bunch of free functions on a single namespace
+
+### Including headers
+
+Using angle brackets `#include <header>` is prefered over double quotes `#include "header"`, as it allows to mock a header
```

Comment:
It should be preferred the faster option for the preprocessor. If the path is relative, we should use `#include "common/Types.hpp"`. If the path is not relative, we should use `#include <fastdds/dds/core/Types.hpp>`.

I can understand that a relative path with `..` is not liked to be used because visual prettiness, like `#include "../rtps/Common.hpp"`. Although my hand will not shake writing that. All for help to the speed of compilation.

And in case a `#include "header"` interferes with a mock, always we can change it to `#include <header>`.